client/contract.go

```diff
@@ -25,9 +26,9 @@ func NewContractClient(c ethclient.Client) *ContractClient {
 }
 
 // GetContractCurrency returns the currency for a specific address
-func (c *ContractClient) GetContractCurrency(addr common.Address, erc20 bool) (*ContractCurrency, error) {
+func (c *ContractClient) GetContractCurrency(addr common.Address, erc20 bool) (*types.Currency, error) {
```

Comment:
I have thus far avoid brining any knowledge of `Rosetta` into `client`. This breaks that abstraction.

Reply:
I think we should instead add a function in `mapper` that wraps this and returns the `Currency`. The simplifications in `mapper` are on-track.

Reply:
Still reconciling this opinion with the need to keep an extra type around.

Reply:
Good point, I've provided an alternative way in 656d0cd that respects the abstraction